src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1539,8 +1539,7 @@ private void SpillEvalStack()
             }
         }
 
-        // PROTOTYPE(dataflow): Revisit and determine if this is too much abstraction, or if it's fine as it is.
-        private void PushArray<T>(ImmutableArray<T> array, Func<T, IOperation> unwrapper = null) where T : IOperation
+        private void VisitAndPushArray<T>(ImmutableArray<T> array, Func<T, IOperation> unwrapper = null) where T : IOperation
```

Comment:
Consider making this an IList or `TList where TList : IList<T>` to avoid unnecessary allocations.

Reply:
> Consider making this an IList or ```TList where TList : IList<T>``` to avoid unnecessary allocations.

I think majority of consumers have an immutable array at hand. IList will make foreach loops less efficient. In any case, I don't think this is critical to address before merging to the parent branch. 

---
In reply to: [194126111](https://github.com/dotnet/roslyn/pull/27583#discussion_r194126111) [](ancestors = 194126111)

Reply:
I don't think this is blocking either.